src/common/api/worker/rest/EntityRestClient.ts

```diff
@@ -49,13 +49,36 @@ export interface EntityRestClientUpdateOptions {
 }
 
 /**
- * Whether to use the cache to fetch the entity
+ * Determines how to handle caching behavior (i.e. reading/writing).
+ *
+ * Use {@link getCacheModeBehavior} to programmatically check the behavior of the cache mode.
  */
 export const enum CacheMode {
 	/** Prefer cached value if it's there or fall back to network. */
 	Cache,
-	/** Prefer the value from network, do not fetch from cache. The entity will still be cached upon loading. */
+	/**
+	 * Prefer the value from network, do not fetch from cache. The entity will still be cached upon loading.
+	 *
+	 * NOTE: This cannot be used for ranged requests.
+	 */
 	Bypass,
+	/** Prefer cached value, but in case of a cache miss, retrieve the value from network but don't write it to cache. */
+	ReadOnly,
+}
+
+/**
+ * Get the behavior of the cache mode for the options
+ * @param cacheMode cache mode to check, or check the default cache mode ({@link CacheMode.Cache})
+ */
+export function getCacheModeBehavior(cacheMode?: CacheMode): { readsFromCache: boolean; writesToCache: boolean } {
```

Comment:
I would do one of
 - `cacheMode: CacheMode | undefined` (so that it's required to pass something)
 - `cacheModel: CacheMode = CacheMode.Cache`